src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -1286,21 +1286,148 @@ export default class DataTable extends Vue {
     }
   }
 
+  _printHead() {
+    return (
+      <thead>
+        <tr>
+          {Object.keys(this.data.head).map((column: string) => {
+            return <th>{this.data.head[column].label}</th>;
+          })}
+        </tr>
+      </thead>
+    );
+  }
+
+  _printFooter() {
+    const resultsCount =
+      this.mode === DataTableModes.CLIENT
+        ? this._showSelectedOnly
+          ? this.selectedRows.length
+          : this.data.body.length
+        : this.config.pagination.results;
+
+    return (
+      <tfoot>
+        <tr>
+          <td colspan={Object.keys(this.data.head).length}>{resultsCount} results</td>
+        </tr>
+      </tfoot>
+    );
+  }
+
+  _printBody() {
+    if (this.data.body.length > 0) {
+      return (
+        <tbody>
+          {this._serializedDataBody.map((bodyRow: DataTableRow) => {
```

Comment:
What if the user applied sorting before printing? Did you check the print functionality in that case?
`this._serializedDataBody` is an array with all the processed rows with assigned IDs, etc but it does not reflect the sorting.
Please, check if `this._sortedData` exists before using `this._serializedDataBody`.